2.2.3: `install` target fails and missing .pc file

Looks like cmake fails in `install` target
```console
+ cd miniupnpc-2.2.3
+ DESTDIR=/home/tkloczko/rpmbuild/BUILDROOT/miniupnpc-2.2.3-2.fc35.x86_64
+ /usr/bin/cmake --install x86_64-redhat-linux-gnu
-- Install configuration: "RelWithDebInfo"
-- Installing: /home/tkloczko/rpmbuild/BUILDROOT/miniupnpc-2.2.3-2.fc35.x86_64/usr/lib64/libminiupnpc.so.2.2.3
-- Installing: /home/tkloczko/rpmbuild/BUILDROOT/miniupnpc-2.2.3-2.fc35.x86_64/usr/lib64/libminiupnpc.so.17
-- Installing: /home/tkloczko/rpmbuild/BUILDROOT/miniupnpc-2.2.3-2.fc35.x86_64/usr/lib64/libminiupnpc.so
-- Installing: /home/tkloczko/rpmbuild/BUILDROOT/miniupnpc-2.2.3-2.fc35.x86_64/usr/lib64/cmake/miniupnpc/libminiupnpc-shared.cmake
-- Installing: /home/tkloczko/rpmbuild/BUILDROOT/miniupnpc-2.2.3-2.fc35.x86_64/usr/lib64/cmake/miniupnpc/libminiupnpc-shared-relwithdebinfo.cmake
CMake Error at x86_64-redhat-linux-gnu/cmake_install.cmake:110 (file):
  file INSTALL cannot find
  "/home/tkloczko/rpmbuild/BUILD/miniupnpc-2.2.3/miniupnpc.h": No such file
  or directory.
```
cmake as weel do not generate and install miniupnpc.pc file (main Makegule is no longer able to generate that file)
```console
[tkloczko@ss-desktop miniupnpc-2.2.3]$ make miniupnpc.pc
make: *** No rule to make target 'miniupnpc.pc'.  Stop.
```

@miniupnp It still complains about a missing header:
```
CMake Error at cmake_install.cmake:131 (file):
  file INSTALL cannot find
  "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/miniupnpc-2.2.3/miniupnpc.h":
  No such file or directory.


make: *** [install/fast] Error 1
```

I found it is possible to disable rpath by pass `-D CMAKE_SKIP_RPATH=ON`. It looks like it is part of the standard cmake modules however I think that it would be good to disable that by default.

I've tested as well https://github.com/miniupnp/miniupnp/pull/591 and it looks good.

It is yet another small issue 😃 
Despite use `-D UPNPC_BUILD_TESTS=ON` ctest is not able to find any units.
```console
[tkloczko@devel-g2v x86_64-redhat-linux-gnu]$ cmake -L |grep TEST
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/home/tkloczko/rpmbuild/BUILD/miniupnp-miniupnpd_2_3_0/miniupnpc/x86_64-redhat-linux-gnu" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
UPNPC_BUILD_TESTS:BOOL=ON
[tkloczko@devel-g2v x86_64-redhat-linux-gnu]$ ctest -vv
Test project /home/tkloczko/rpmbuild/BUILD/miniupnp-miniupnpd_2_3_0/miniupnpc/x86_64-redhat-linux-gnu
No tests were found!!!
```

--- a/miniupnpc/CMakeLists.txt
+++ b/miniupnpc/CMakeLists.txt
@@ -219,6 +219,8 @@ if (UPNPC_BUILD_TESTS)
 # set (UPNPC_INSTALL_TARGETS ${UPNPC_INSTALL_TARGETS} testminixml minixmlvalid testupnpreplyparse testigddescparse testminiwget)
 endif ()
 
+configure_file(miniupnpc.pc.in miniupnpc.pc @ONLY)
+
 if (NOT UPNPC_NO_INSTALL)
   install (FILES
     include/miniupnpc.h
@@ -236,6 +238,10 @@ if (NOT UPNPC_NO_INSTALL)
 
   install(FILES miniupnpc-config.cmake
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/miniupnpc
+  )
+
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/miniupnpc.pc
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
   )
 endif()
 

--- a/miniupnpc/CMakeLists.txt
+++ b/miniupnpc/CMakeLists.txt
@@ -1,11 +1,12 @@
-cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
 
-project (miniupnpc C)
+project (miniupnpc
+         VERSION 2.2.3
+         DESCRIPTION "UPnP IGD client lightweight library"
+         HOMEPAGE_URL https://miniupnp.tuxfamily.org/
+         LANGUAGES C)
 
-set (MINIUPNPC_VERSION 2.2.3)
 set (MINIUPNPC_API_VERSION 17)
-set (MINIUPNPC_DESCRIPTION "UPnP IGD client lightweight library")
-set (MINIUPNPC_HOMEPAGE_URL https://miniupnp.tuxfamily.org/)
 
 option (UPNPC_BUILD_STATIC "Build static library" TRUE)
 option (UPNPC_BUILD_SHARED "Build shared library" TRUE)
@@ -143,7 +144,7 @@ if (UPNPC_BUILD_SHARED)
   add_library (miniupnpc::miniupnpc ALIAS libminiupnpc-shared)
   set_target_properties (libminiupnpc-shared PROPERTIES EXPORT_NAME miniupnpc)
   set_target_properties (libminiupnpc-shared PROPERTIES OUTPUT_NAME "miniupnpc")
-  set_target_properties (libminiupnpc-shared PROPERTIES VERSION ${MINIUPNPC_VERSION})
+  set_target_properties (libminiupnpc-shared PROPERTIES VERSION ${PROJECT_VERSION})
   set_target_properties (libminiupnpc-shared PROPERTIES SOVERSION ${MINIUPNPC_API_VERSION})
   target_link_libraries (libminiupnpc-shared PRIVATE miniupnpc-private)
   target_compile_definitions(libminiupnpc-shared PRIVATE MINIUPNP_EXPORTS)

--- a/miniupnpc/CMakeLists.txt
+++ b/miniupnpc/CMakeLists.txt
@@ -4,6 +4,8 @@ project (miniupnpc C)
 
 set (MINIUPNPC_VERSION 2.2.3)
 set (MINIUPNPC_API_VERSION 17)
+set (MINIUPNPC_DESCRIPTION "UPnP IGD client lightweight library")
+set (MINIUPNPC_HOMEPAGE_URL https://miniupnp.tuxfamily.org/)
 
 option (UPNPC_BUILD_STATIC "Build static library" TRUE)
 option (UPNPC_BUILD_SHARED "Build shared library" TRUE)

--- a/miniupnpc/Makefile
+++ b/miniupnpc/Makefile
@@ -3,7 +3,7 @@
 # http://miniupnp.free.fr/
 # https://miniupnp.tuxfamily.org/
 # https://github.com/miniupnp/miniupnp
-# (c) 2005-2021 Thomas Bernard
+# (c) 2005-2022 Thomas Bernard
 # to install use :
 # $ make DESTDIR=/tmp/dummylocation install
 # or
@@ -293,6 +293,7 @@ $(BUILD)/miniupnpc.pc:	VERSION
 	echo "" >> $@
 	echo "Name: miniUPnPc" >> $@
 	echo "Description: UPnP IGD client lightweight library" >> $@
+	echo "URL: https://miniupnp.tuxfamily.org/" >> $@
 	echo "Version: $(VERSION)" >> $@
 	echo "Libs: -L\$${libdir} -lminiupnpc" >> $@
 	echo "Cflags: -I\$${includedir}" >> $@

--- a/miniupnpc/miniupnpc.pc.in
+++ b/miniupnpc/miniupnpc.pc.in
@@ -1,0 +1,18 @@
+# this template is filled-in by CMake `configure_file(... @ONLY)`
+# the `@....@` are filled in by CMake configure_file(), 
+# from variables set in your CMakeLists.txt or by CMake itself
+#
+# Good tutoral for understanding .pc files: 
+# https://people.freedesktop.org/~dbn/pkg-config-guide.html
+
+prefix="@CMAKE_INSTALL_PREFIX@"
+exec_prefix="${prefix}"
+libdir="${prefix}/lib"
+includedir="${prefix}/include"
+
+Name: @PROJECT_NAME@
+Description: @CMAKE_PROJECT_DESCRIPTION@
+URL: @CMAKE_PROJECT_HOMEPAGE_URL@
+Version: @PROJECT_VERSION@
+Libs: -L"${libdir}" -lminiupnpc
+Cflags: -I"${includedir}"

--- a/miniupnpc/miniupnpc.pc.in
+++ b/miniupnpc/miniupnpc.pc.in
@@ -11,8 +11,8 @@ libdir="${prefix}/lib"
 includedir="${prefix}/include"
 
 Name: @PROJECT_NAME@
-Description: @MINIUPNPC_DESCRIPTION@
-URL: @MINIUPNPC_HOMEPAGE_URL@
-Version: @MINIUPNPC_VERSION@
+Description: @PROJECT_DESCRIPTION@
+URL: @PROJECT_HOMEPAGE_URL@
+Version: @PROJECT_VERSION@
 Libs: -L"${libdir}" -lminiupnpc
 Cflags: -I"${includedir}"

--- a/miniupnpc/miniupnpc.pc.in
+++ b/miniupnpc/miniupnpc.pc.in
@@ -11,8 +11,8 @@ libdir="${prefix}/lib"
 includedir="${prefix}/include"
 
 Name: @PROJECT_NAME@
-Description: @CMAKE_PROJECT_DESCRIPTION@
-URL: @CMAKE_PROJECT_HOMEPAGE_URL@
-Version: @PROJECT_VERSION@
+Description: @MINIUPNPC_DESCRIPTION@
+URL: @MINIUPNPC_HOMEPAGE_URL@
+Version: @MINIUPNPC_VERSION@
 Libs: -L"${libdir}" -lminiupnpc
 Cflags: -I"${includedir}"

